R/module-visualize-network-server.R

```diff
@@ -0,0 +1,176 @@
+            container: document.getElementById('network-cy'),
+            elements: [", paste(elements, collapse = ", "), "],
+            style: [
+                {
+                    selector: 'node',
+                    style: {
+                        'background-color': '#66ccff',
+                        'label': 'data(label)'
+                    }
+                },
+                {
+                    selector: 'edge',
+                    style: {
+                        'width': 3,
+                        'line-color': '#ccc',
+                        'label': 'data(interaction)',
+                        'curve-style': 'bezier',
+                        'target-arrow-shape': 'triangle',
+                        'target-arrow-color': '#234',
+                        'edge-offset': 10,
+                        'text-margin-y': -10,  // Adjust the vertical distance of edge labels
+                        'text-halign': 'center',  // Horizontally align labels to the center
+                        'edge-text-rotation': 'autorotate'
+                    }
+                }
+            ],
+            layout: {
+                name: 'dagre',
+                rankDir: 'TB',
+                animate: true,
+                fit: true
+            }
+        });
+        // Capture the event when an edge is clicked
+        cy.on('tap', 'edge', function(evt) {
+            var edge = evt.target;
+            const edgeId = edge.id();
+            Shiny.setInputValue('network-edgeClicked', { id: edge.id() });
+        });
+        ")
+        # Return the JavaScript code to be executed
+        return(list(js_code = js_code,
+            edges_table = subnetwork$edges,
+            nodes_table = subnetwork$nodes
+        ))
+    })
+
+    # Observe the button click to trigger network rendering and table display
+    observeEvent(input$showNetwork, {
+        # Ensure that all required inputs are valid before rendering the network
+        req(df(), proteinIdType(), pValue(), logFC())
+        
+        # When the "Display Network" button is clicked, run the network rendering logic
+        render_data <- renderNetwork()
+        js_code <- render_data$js_code
+        edges_table <- render_data$edges_table
+        nodes_table <- render_data$nodes_table
+        
+        # Send the JavaScript code to the frontend to render Cytoscape.js
+        session$sendCustomMessage(type = 'runCytoscape', message = js_code)
+
+        output$nodesTable <- renderDT({
+            datatable(nodes_table, options = list(pageLength = 10, searchable = TRUE))
+        })
+        
+        output$edgesTable <- renderDT({
+            datatable(edges_table, options = list(pageLength = 10, searchable = TRUE), selection = 'single')
+        })
+
+    })
+
+    # Observe the event when an edge is clicked
+    observeEvent(input$edgeClicked, {
+        edge_id <- input$edgeClicked$id
+        edge_parts <- strsplit(edge_id, "-")[[1]]
```

Comment:
rather than a string split, I believe we already have source, target, and interaction data stored in the edge as well?